packages/js/src/ui/components/PopoverWithBell.tsx

```diff
@@ -0,0 +1,20 @@
+import { createSignal, useContext } from 'solid-js';
+import { Bell, Popover } from '.';
+
+export const PopoverWithBell = (props: any) => {
```

Comment:
I generally don't think we should have internal reusable components. I think that the logic here will just be in the component that we export from the package itself. Will we be exporing a `<Popover />` ? If so, would the API be :
```
<Inbox>
  <Popover>
    <PopoverTrigger>
      <Bell />
    </PopoverTrigger>
    <PopoverContent>
      <Foo/>
    </PopoverContent>
  </Popover>
</Inbox>
  ```
  or something similar?

Reply:
The reason to not have internal reusable components is because we will have to prop drill a lot of descriptors in a big reusable internal component. It's better to have components that map to one element in the DOM, or compound components for internal consumption.

Reply:
Yes, composition only, avoid props.